demo/src/main/java/com/hazelcast/demo/AllTest.java

```diff
@@ -25,7 +25,7 @@
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.core.IMap;
 import com.hazelcast.internal.diagnostics.HealthMonitorLevel;
-import com.hazelcast.query.SqlPredicate;
+import com.hazelcast.query.impl.predicates.SqlPredicate;
```

Comment:
The idea is that users should not instantiate the predicates directly (using internal classes from `query.impl`). Instead, they should use the factory methods provided in `query.Predicates`. For the SQL predicate, there is `Predicates.sql()`, and there are also factory methods for the paging predicate, the partition predicate etc.

So I suggest to replace all direct instantiations of the impl classes with calls to the appropriate `Predicates` factory methods.

Reply:
Done.